addons/hr_expense/hr_expense.py

```diff
@@ -385,94 +311,154 @@ def action_view_move(self, cr, uid, ids, context=None):
+    sequence = fields.Integer()
+    employee_id = fields.Many2one('hr.employee', string="Employee", readonly=True, states={'draft': [('readonly', False)]}, required=True, default=lambda self: self._employee_get())
+    state = fields.Selection([
+        ('draft', 'New'),
+        ('confirm', 'Submitted'),
+        ('accepted', 'Approved'),
+        ('cancel', 'Cancelled'),
+        ],
+        string='Status', readonly=True, track_visibility='onchange', copy=False, required=True, default='draft')
+    expense_tax_id = fields.Many2many('account.tax', 'expense_line_tax', 'expense_line_id', 'tax_id', string='Taxes', readonly=True, states={'draft': [('readonly', False)]})
+
+    _order = "sequence, date"
+
+    @api.onchange('product_id')
+    def onchange_product_id(self):
+        if self.product_id:
+            self.unit_amount = self.product_id.price_get('standard_price')[self.product_id.id]
+            self.uom_id = self.product_id.uom_id.id
+
+    @api.onchange('product_id', 'uom_id')
+    def onchange_uom(self):
+        res = {'value': {}}
+        if self.product_id:
+            if self.uom_id.category_id.id != self.product_id.uom_id.category_id.id:
+                res['warning'] = {'title': _('Warning'), 'message': _('Selected Unit of Measure does not belong to the same category as the product Unit of Measure')}
+                self.uom_id = self.product_id.uom_id.id
             return res
-        product = self.pool.get('product.product').browse(cr, uid, product_id, context=context)
-        uom = self.pool.get('product.uom').browse(cr, uid, uom_id, context=context)
-        if uom.category_id.id != product.uom_id.category_id.id:
-            res['warning'] = {'title': _('Warning'), 'message': _('Selected Unit of Measure does not belong to the same category as the product Unit of Measure')}
-            res['value'].update({'uom_id': product.uom_id.id})
-        return res
+
+    @api.multi
+    def expense_line_draft(self):
+        return self.write({'state': 'draft'})
+
+    @api.multi
+    def expense_confirm(self):
+        sheet_obj = self.env['hr_expense.sheet']
+        for expense in self:
+            sheet = sheet_obj.search([('state', 'in', ['draft', 'confirm']), ('employee_id', '=', expense.employee_id.id)], limit=1)
+            if not sheet:
+                vals = {'name': ("%s %s" % ('Expenses for', expense.employee_id.name)),
+                        'employee_id': expense.employee_id.id,
+                        'employee_payable_account_id': expense.employee_id.address_home_id.property_account_payable.id,
+                        'department_id': expense.employee_id.department_id.id}
+                expense.write({'expense_id': sheet_obj.create(vals).id, 'state': 'confirm'})
+            else:
+                expense.write({'expense_id': sheet.id, 'state': 'confirm'})
+            expense.expense_id.signal_workflow('confirm')
```

Comment:
@rim-odoo  @mdi-odoo
If expense sheet is in 'draft' state by "set to draft" button and employee responsible 'submit' expenses again so expense sheet is in 'draft' state and we want to directly submit sheet in 'Waiting Approval' state it from employee submit expense. that reason i have manually trigger workflow signals..